src/utils/signer.ts

```diff
@@ -0,0 +1,440 @@
+/*
+ * This is a copied and modified file from https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/Signer/Signer.ts
```

Comment:
Would it worth specifying which point in the commit history of these files that it was copied from? That way we can reference back to it if there are ever changes made by Amplify.

Reply:
Good call! I'll include the actual commit I copied this from.